src/App.tsx

```diff
@@ -5,30 +5,38 @@ import '@fontsource/roboto/700.css';
 
 import './App.css';
 
-import InvoiceList from './components/InvoiceList';
-import Invoice from './components/Invoice';
+import InvoiceList from './components/InvoiceList/InvoiceList';
 
 import { createBrowserRouter, RouterProvider } from 'react-router-dom';
 import Layout from './components/Layout';
 import { LocalizationProvider } from '@mui/x-date-pickers/LocalizationProvider';
 import { AdapterDateFns } from '@mui/x-date-pickers/AdapterDateFns';
+import { QueryClient, QueryClientProvider } from '@tanstack/react-query';
+import { ReactQueryDevtools } from '@tanstack/react-query-devtools';
+import React from 'react';
+import CreateInvoice from './components/Invoice/CreateInvoice';
+import EditInvoice from './components/Invoice/EditInvoice';
 
 const router = createBrowserRouter([
   {
     element: <Layout />,
     children: [
       { path: '', element: <InvoiceList /> },
-      { path: 'add', element: <Invoice /> },
-      { path: 'edit/:id', element: <Invoice /> },
+      { path: 'add', element: <CreateInvoice /> },
+      { path: 'edit/:id', element: <EditInvoice /> },
     ],
   },
 ]);
 
 function App() {
+  const [queryClient] = React.useState(() => new QueryClient());
```

Comment:
I would rather create a query client outside the component, so always only one instance of it is available.

Reply:
If I put it out of the App function the app does not compile

Reply:
corrected